var/spack/repos/builtin/packages/py-opppy/package.py

```diff
@@ -11,21 +11,29 @@ class PyOpppy(PythonPackage):
     analysis library designed to extract, store, and plot information from
     output and dump files generated by scientific software packages."""
 
-    homepage = "https://github.com/lanl/opppy"
-    url = "https://github.com/lanl/OPPPY/archive/opppy-0_1_2.tar.gz"
-    git = "https://github.com/lanl/opppy.git"
+    homepage = "https://github.com/lanl/OPPPY"
+    url = "https://github.com/lanl/OPPPY/archive/refs/tags/opppy-0_1_2.tar.gz"
+    git = "https://github.com/lanl/OPPPY.git"
     maintainers("clevelam")
 
     version("master", branch="master")
+    version("0_1_7", sha256="6ef17e289de2c469ac931288fc5debee77583b300e35fa72080ee06d0e27f7af")
+    version("0_1_6", sha256="1f385ffa39c90d3b338aceb563c5f4952d01041fff0ec94084c84e9e1107023e")
     version("0_1_5", sha256="d9df166d347c18d4f145059b4c2fb23dbfbecf0dd5a3398f29e52d3e261844b0")
     version("0_1_4", sha256="22d81a64856f4c12f8079440c837d7d1f45153e68c405b45bed8b6d35831e948")
     version("0_1_3", sha256="c3ca97f2ff8ab319b5c7257baa8cab852387dc00d426b4534c06f0894363c541")
     version("0_1_2", sha256="ef3795d3164fa0aa7ea7da7e223d6d0a48d2960aefd03a7d90cdb8b8f480cd4c")
     version("0_1_1", sha256="505c023853e75552abc65de9777a125ecb6a99a1cb4e605a4f702af837e3168b")
```

Comment:
This works and, by that, I mean I re-confirmed the branch and 7 version sha256.  IMO since only the oldest listed version has a different URL, you could simplify this by getting rid of lines 33-39 and using the following instead:

```suggestion
    version(
        "0_1_1", 
        sha256="505c023853e75552abc65de9777a125ecb6a99a1cb4e605a4f702af837e3168b",
        url="https://github.com/lanl/OPPPY/archive/refs/tags/OPPPY-0_1_1.tar.gz",
    )
```

Reply:
If you're an OPPPY developer, you could also rename the old tarball, although that's somewhat frowned upon in case someone was using the old URL.

Reply:
I am an OPPPY developer and I did consider changing this single tag, but decided against it for continuity sake. I do like the simpler url fix. My first thought was to try this but I wasn't sure it was an option. I'll make this change and push it shortly. Thanks!